### PR TITLE
Add bundle upgrade UI to subscription management view

### PR DESCRIPTION
**Pull request by lesleyjanenorton**
## Description

This PR:
- Adds "bundleUpgrade" feature
- Adds bundle upgrade UI to in-app subscription management view. This UI is hidden unless:
   -  the bundleUpgrade feature is enabled
   -  the subscription type is "web"
   -  the locale is en_US or en_CA

Screen Shot 2022-09-01 at 5 53 47 PM

## Reference
VPN-2726, #4243 

## Checklist
    
- [ ] My code follows the style guidelines for this project
- [ ] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [ ] I have performed a self review of my own code
- [ ] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

### CODE REVIEW

File: src/constants.h [hunk 1]
```diff
@@ -119,6 +119,15 @@ PRODBETAEXPR(
     "97e8ba9cf12fb3de53cc42a4e6577ed64df493c247b414fea036818d3823560e",
     "3c01446abe9036cea9a09acaa3a520ac628f20a7ae32ce861cb2efb70fa0c745");
 
+PRODBETAEXPR(const char*, relayUrl, "https://relay.firefox.com",
+             "https://stage.fxprivaterelay.nonprod.cloudops.mozgcp.net");
+
+PRODBETAEXPR(const char*, bundleUpgradeUrl,
```
Comment (lesleyjanenorton) [inline src/constants.h:125]: This makes me nervous. Would it be better to bounce folks through Guardian and let Guardian redirect?

**Review by flodolo — APPROVED** [review]

**Review by bakulf — APPROVED** [review]

File: src/featureslist.h [hunk 2]
```diff
@@ -50,6 +50,14 @@ FEATURE_SIMPLE(appReview,      // Feature ID
                QStringList(),  // feature dependencies
                FeatureCallback_iosOrAndroid)
 
+FEATURE_SIMPLE(bundleUpgrade,     // Feature ID
+               "Bundle Upgrade",  // Feature name
+               "2.10",            // released
+               true,              // Can be flipped on
+               true,              // Can be flipped off
+               QStringList(),     // feature dependencies
+               FeatureCallback_true)
```
Comment (bakulf) [inline src/featureslist.h:59]: we want it false by default ,right?

File: src/ui/screens/settings/ViewSubscriptionManagement/SubscriptionManagementItem.qml [hunk 3]
```diff
@@ -82,7 +119,7 @@ ColumnLayout {
         spacing: VPNTheme.theme.listSpacing
 
         Layout.alignment: Qt.AlignVCenter
-        Layout.bottomMargin: VPNTheme.theme.listSpacing
+        Layout.bottomMargin:  VPNTheme.theme.listSpacing
```
Comment (flozia) [inline src/ui/screens/settings/ViewSubscriptionManagement/SubscriptionManagementItem.qml:122]: Nit: An unneeded space sneaked in here.

File: src/ui/screens/settings/ViewSubscriptionManagement/ViewSubscriptionManagement.qml [hunk 4]
```diff
@@ -159,7 +159,7 @@ VPNViewBase {
                     VPNSubscriptionData.planCurrency,
                     VPNSubscriptionData.planAmount,
                 ),
-                type: "text",
+                type: (Qt.locale().name === "en_US" || Qt.locale().name === "en_CA" ) && VPNFeatureList.get("bundleUpgrade").isSupported ? "text-upgrade" : "text",
```
Comment (flozia) [inline src/ui/screens/settings/ViewSubscriptionManagement/ViewSubscriptionManagement.qml:162]: Question: Checking on the locale is only a temporary solution?

**Review by flozia — COMMENTED** [review]
Nice! The bundle upgrade is currently enabled by default. Is this intended?

File: src/ui/screens/settings/ViewSubscriptionManagement/ViewSubscriptionManagement.qml (hunk 4 above)
Comment (lesleyjanenorton) [reply src/ui/screens/settings/ViewSubscriptionManagement/ViewSubscriptionManagement.qml:162]: semi-permanent - we're limiting this to various "en" locales for now (and unfortunately). this stipulation is based on geographic availability of the relay phone masking feature.

File: src/ui/screens/settings/ViewSubscriptionManagement/SubscriptionManagementItem.qml (hunk 3 above)
Comment (lesleyjanenorton) [reply src/ui/screens/settings/ViewSubscriptionManagement/SubscriptionManagementItem.qml]: Those sneaky spaces are always up to no good

File: glean/metrics.yaml [hunk 5]
```diff
@@ -440,6 +440,58 @@ sample:
       - amarchesini@mozilla.com
     expires: never
 
+  bundle_upsell_link_clicked:
```
[the remark this answers is not in the document]
Comment (lesleyjanenorton) [reply glean/metrics.yaml:443]: yep, the bug is here: https://bugzilla.mozilla.org/show_bug.cgi?id=1789284

File: src/featureslist.h (hunk 2 above)
Comment (lesleyjanenorton) [reply src/featureslist.h]: done

**Comment by lesleyjanenorton** [comment]
@flozia 

> Nice! The bundle upgrade is currently enabled by default. Is this intended?

Fixed - disabled by default now.

**Pull request merged**